modules/curl/8.8.0/patches/add_build_file.patch

```diff
@@ -0,0 +1,317 @@
++    "-DHAVE_VARIADIC_MACROS_GCC=1",
++    "-DHAVE_WRITABLE_ARGV=1",
++    "-DHAVE_WRITEV=1",
++    "-DRECV_TYPE_ARG1=int",
++    "-DRECV_TYPE_ARG2=void*",
++    "-DRECV_TYPE_ARG3=size_t",
++    "-DRECV_TYPE_ARG4=int",
++    "-DRECV_TYPE_RETV=ssize_t",
++    "-DRETSIGTYPE=void",
++    "-DSELECT_QUAL_ARG5=",
++    "-DSELECT_TYPE_ARG1=int",
++    "-DSELECT_TYPE_ARG234=fd_set*",
++    "-DSELECT_TYPE_RETV=int",
++    "-DSEND_QUAL_ARG2=const",
++    "-DSEND_TYPE_ARG1=int",
++    "-DSEND_TYPE_ARG2=void*",
++    "-DSEND_TYPE_ARG3=size_t",
++    "-DSEND_TYPE_ARG4=int",
++    "-DSEND_TYPE_RETV=ssize_t",
++    "-DSIZEOF_CURL_OFF_T=8",
++    "-DSIZEOF_INT=4",
++    "-DSIZEOF_LONG=8",
++    "-DSIZEOF_OFF_T=8",
++    "-DSIZEOF_SHORT=2",
++    "-DSIZEOF_SIZE_T=8",
++    "-DSIZEOF_TIME_T=8",
++    "-DSTDC_HEADERS=1",
++    "-DSTRERROR_R_TYPE_ARG3=size_t",
++    "-DTIME_WITH_SYS_TIME=1",
++    "-DUSE_THREADS_POSIX=1",
++    "-DUSE_UNIX_SOCKETS=1",
++
++    # Extra defines needed by curl
++    "-DBUILDING_LIBCURL",
++    "-DCURL_HIDDEN_SYMBOLS",
++]
++
++cc_library(
++    name = "curl",
++    srcs = glob([
++        "lib/**/*.c",
++        "lib/**/*.h",
++    ]),
++    hdrs = glob(["include/curl/*.h"]),
++    copts = select({
++        "@platforms//os:windows": [
++            "/DWIN32",
++            "/DBUILDING_LIBCURL",
++            "/DUSE_WIN32_IDN",
++            "/DWANT_IDN_PROTOTYPES",
++            "/DUSE_IPV6",
++            "/DUSE_WINDOWS_SSPI",
++            "/DUSE_SCHANNEL",
++        ],
++        "@platforms//os:macos": _BASE_CURL_COPTS,
++        "//conditions:default": _BASE_CURL_COPTS + [
++            "-DHAVE_LINUX_TCP_H=1",
++            "-DHAVE_MSG_NOSIGNAL=1",
++        ],
++    }) + [
++        "-DOS=\"os\"",
++    ],
++    defines = ["CURL_STATICLIB"],
++    includes = [
++        "include/",
++        "lib/",
++    ],
++    linkopts = select({
++        "@platforms//os:windows": [
++            "-DEFAULTLIB:ws2_32.lib",
++            "-DEFAULTLIB:advapi32.lib",
++            "-DEFAULTLIB:crypt32.lib",
++            "-DEFAULTLIB:Normaliz.lib",
++        ],
++        "@platforms//os:macos": [
++            "-Wl,-framework,SystemConfiguration",
++            "-lpthread",
++        ],
++        "//conditions:default": [
++            "-lpthread",
++        ],
++    }),
++    local_defines = select({
++        "@platforms//os:macos": [
++            'CURL_CA_BUNDLE=\\"/etc/ssl/cert.pem\\"',
++            'CURL_CA_PATH=\\"/etc/ssl/certs\\"',
++        ],
++        "@platforms//os:linux": [
++            'CURL_CA_BUNDLE=\\"/etc/ssl/certs/ca-certificates.crt\\"',
++            'CURL_CA_PATH=\\"/etc/ssl/certs\\"',
++        ],
++        "//conditions:default": [],
++    }) + select({
++        ":use_mbedtls_setting": ["USE_MBEDTLS"],
++        "//conditions:default": [],
++    }) + select({
++        ":http_only_setting": ["HTTP_ONLY"],
++        "//conditions:default": [],
++    }) + select({
++        ":ca_bundle_is_unset": [],
```

Comment:
shouldn't the defaults go here? Or are later defines guaranteed to override earlier defines?

Reply:
these being later make sure they override, but i think the problem is i have to do it this way because i cannot make a config_setting that is for "not unset" ? and i can't add a select() in the default value of the build setting. I could write a simple rule in this repo to fix this, but felt like overkill. this does produce a warning since it's overridden when you pass it, but i figured i could fix that if folks cared, or ever even passed this flag

Reply:
do nested selects work? something like

```bzl
select({
    ":ca_bundle_is_unset": select({
        "@platforms//os:macos": ['CURL_CA_BUNDLE=\\"/etc/ssl/cert.pem\\"'],
        "@platforms//os:linux": ['CURL_CA_PATH=\\"/etc/ssl/certs\\"'],
        "//conditions:default": [],
    }),
    "//conditions:default": ['CURL_CA_BUNDLE=\\"$(CURL_CA_BUNDLE)\\"'],
})
```

Reply:
do nested selects ever work? in this case I at least get:

```
ERROR: package contains errors: : Traceback (most recent call last):
        File "/Users/ksmiley/dev/curl/BUILD.bazel", line 284, column 8, in <toplevel>
                }) + select({
Error: Cannot combine incompatible types (select of list, select of select)
```

If I removed all the other selects I'm appending I get:

```
ERROR: /Users/ksmiley/dev/curl/BUILD.bazel:223:11: //:curl: expected value of type 'list(string)' for each branch in select expression of attribute 'local_defines' in 'cc_library' rule (including '//:ca_bundle_is_unset'), but got select({"@platforms//os:macos": ["CURL_CA_BUNDLE=\\\"/etc/ssl/cert.pem\\\"", "CURL_CA_PATH=\\\"/etc/ssl/certs\\\""], "@platforms//os:linux": ["CURL_CA_BUNDLE=\\\"/etc/ssl/certs/ca-certificates.crt\\\"", "CURL_CA_PATH=\\\"/etc/ssl/certs\\\""], "//conditions:default": []}) (select)
```

Reply:
ok, thanks for trying.

I think the "correct" way to do this is to make a `:ca_bundle_is_unset` `config_setting` per-OS -- so `:ca_bundle_is_unset_macos` which has `flag_values = {":ca_bundle": ""}` and `constraint_values = ["@platform//os:macos"]`, and same with linux. Then use one single select. But as you can tell, I'm not very well-versed in configurability, so what you have is good enough for me :)

Reply:
oh very cool idea. I did that and it worked. probably not worth another version unless folks are annoyed. i will do that for the next one. https://github.com/keith/curl/commit/f83a92f0893a694c1d38f1379f60e56358db923a